jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js

```diff
@@ -39,15 +45,48 @@ YUI.add('deployment-bar', function() {
     */
     getInitialState: function() {
       return {
+        hasCommits: false,
         latestChangeDescription: null
       };
     },
 
+    componentDidMount: function() {
+      this._updateHasCommits();
+    },
+
     componentWillReceiveProps: function(nextProps) {
+      this._updateHasCommits();
       this._updateLatestChange(nextProps.currentChangeSet);
     },
 
     /**
+      Check if we have any commits.
+
+      @param {Function} callback A function to call once the state has updated.
+      @method _updateHasCommits
+    */
+    _updateHasCommits: function() {
+      var hasCommits = false;
+      if (!this.state.hasCommits) {
+        this.props.services.forEach(service => {
```

Comment:
Because this is called from `componentWillReceiveProps` shouldn't it also take a props argument and use that one instead? Because here you're using `currentProps` instead of `nextProps` in the `componentWillReceiveProps` case.